scene/2d/animated_sprite_2d.cpp

```diff
@@ -154,32 +154,26 @@ void AnimatedSprite2D::_validate_property(PropertyInfo &p_property) const {
 void AnimatedSprite2D::_notification(int p_what) {
 	switch (p_what) {
 		case NOTIFICATION_INTERNAL_PROCESS: {
-			if (frames.is_null()) {
-				return;
-			}
-			if (!frames->has_animation(animation)) {
-				return;
-			}
-			if (frame < 0) {
+			if (frames.is_null() || !frames->has_animation(animation) || frame < 0) {
 				return;
 			}
 
 			double remaining = get_process_delta_time();
 
 			while (remaining) {
-				double speed = frames->get_animation_speed(animation) * speed_scale;
+				double speed = frames->get_animation_speed(animation) * abs(speed_scale);
 				if (speed == 0) {
 					return; // Do nothing.
 				}
 
 				if (timeout <= 0) {
 					timeout = _get_frame_duration();
 
-					int fc = frames->get_frame_count(animation);
-					if ((!backwards && frame >= fc - 1) || (backwards && frame <= 0)) {
+					int ending_frame = frames->get_frame_count(animation) - 1;
+					if ((!backwards && frame >= ending_frame) || (backwards && frame <= 0)) {
 						if (frames->get_animation_loop(animation)) {
 							if (backwards) {
-								frame = fc - 1;
+								frame = ending_frame;
```

Comment:
Since the backwards branching is done repeatedly, it should be `!backwards` first instead of `!backwards && frame >= ending_frame` to reduce the number of if checks.

Reply:
I was considering changing these messy conditions outright to be much simpler at the cost of sliiiiightly more memory usage, but that can be done another time.

Reply:
I think this repeated branching is simply not readable and needs to be changed.

Reply:
I don't have the guarantee this PR will be merged but I can give a try at it later.

Reply:
Yep, I think it could be integrated with AnimatedSprite3D spped_scale PR.

Reply:
Yeah, makes sense. That one is guaranteed. I can update [that PR](https://github.com/godotengine/godot/pull/64155) to harmonise both classes (make their internals look almost alike) and hopefully look nicer, then this one can be rebased and considered if necessary.